### PR TITLE
Document release notes procedure

### PR DESCRIPTION
**Pull request by GitHK**
Adds documentation how to handle the release notes.
It is important that the procedure is written down, since it's the scrum master's responsibility to enforce it.
This also makes it easier to update.

I have tried to create as much automation as possible in order to aid the scrum master.

Bonus:
- upgraded old not working `.pre-commit-config.yaml` config
- updated user readme in `scripts`
- moved some code to the shared utils module inside `scripts`

TODOs:
- [x] before committing run entire readme once

Related issues:
- closes https://github.com/ITISFoundation/osparc-issues/issues/1637

### CODE REVIEW

File: release-notes/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```
+- `<VERSION>` format: `X.X.X` where `X` is a number
+- `<DATE>` format: `dd.mm` where `dd` is a valid caldendar day and `mm` is valid calendar month
+- `<TIME>` format: `HH.MM` where `HH` is a valid hour and `MM` is valid minute
+
+
+:warning: Do not forget to commit and push the changes in the reposity! :rotating_light:
+
+
+### 5. `staging deplyed`
```
Comment (pcrespov) [inline release-notes/README.md:84]: deployed

GENERAL NOTE: pass a spellchecker to this readme :-)

File: release-notes/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+### 1. `review date`
+
+Ask Nik (be insistent) for the review date as soon as possible.
+It is impossible to organise without a release date.
+
+### 2. `pick staging and production release dates`
+
+Ask team members if it is possible to release to:
+- `staging` 1 day before the reivew
```
Comment (pcrespov) [inline release-notes/README.md:39]: review

GENERAL NOTE: pass a spellchecker to this readme :-)

File: release-notes/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Please note, that if you hotfix a change to a production deployment, you have to run the below command and follow it's istructions.
+
+:warning: Do not forget to commit and push the changes in the reposity! :rotating_light:
+
+```shell
+python scripts/run_after_hotfix_to_prod.py <VERSION>
+```
+- `<VERSION>` format: `X.X.X` where `X` is a number
+
+:warning: Do not forget to commit and push the changes in the reposity! :rotating_light:
```
Comment (pcrespov) [inline release-notes/README.md:118]: repository

**Review by pcrespov — COMMENTED** [review]
thx
1. pass a spellchecker to this readme :-)
2. IMO some of the scripts are unnecessary
3. i find the workflow a bit **manual** which IMO reduces the changes to succeed ...

**Review by sanderegg — COMMENTED** [review]
ok, thanks

**Review by matusdrobuliak66 — APPROVED** [review]

**Comment by jsaq007** [comment]
👍🏼

**Pull request merged**